Search/SearchManager.php

```diff
@@ -178,132 +190,229 @@ public function getStatus()
-        $urlField = $metadata->getUrlField();
-        $titleField = $metadata->getTitleField();
-        $descriptionField = $metadata->getDescriptionField();
-        $imageUrlField = $metadata->getImageUrlField();
-        $localeField = $metadata->getLocaleField();
-        $fieldMapping = $metadata->getFieldMapping();
-
-        $accessor = PropertyAccess::createPropertyAccessor();
-
-        $document = $this->factory->makeDocument();
-        $document->setId($accessor->getValue($object, $idField));
-        $document->setClass($metadata->getName());
-
-        if ($urlField) {
-            $url = $accessor->getValue($object, $urlField);
-            if ($url) {
-                $document->setUrl($accessor->getValue($object, $urlField));
+        $classNames = $this->metadataFactory->getAllClassNames();
+        $indexNames = array();
+
+        foreach ($classNames as $className) {
+            $metadata = $this->getMetadataFor($className);
+
+            foreach ($metadata->getIndexMetadatas() as $indexMetadata) {
+                if ($categories) {
+                    if (in_array($indexMetadata->getCategoryName(), $categories)) {
+                        $indexNames[] = $indexMetadata->getIndexName();
+                    }
+                    continue;
+                }
+
+                $indexNames[] = $indexMetadata->getIndexName();
             }
         }
 
-        if ($titleField) {
-            $title = $accessor->getValue($object, $titleField);
-            if ($title) {
-                $document->setTitle($accessor->getValue($object, $titleField));
+        return $indexNames;
+    }
+
+    /**
+     * List all of the expanded index names in the search implementation
+     * optionally only in the given locale.
+     *
+     * @param string $locale
+     * @return string[]
+     */
+    private function getLocalizedIndexNames($locale = null)
+    {
+        $localizedIndexNames = array();
+        $indexNames = $this->getIndexNames();
+
+        foreach ($indexNames as $indexName) {
+            foreach ($this->getLocalizedIndexNamesFor($indexName, $locale) as $localizedIndexName) {
+                $localizedIndexNames[$localizedIndexName] = $localizedIndexName;
             }
         }
 
-        if ($descriptionField) {
-            $description = $accessor->getValue($object, $descriptionField);
-            if ($description) {
-                $document->setDescription($accessor->getValue($object, $descriptionField));
+        return array_values($localizedIndexNames);
+    }
+
+    /**
+     * Retrieve all the index names including localized names (i.e. variants)
+     * for the given index name, optionally limiting to the given locale.
+     *
+     * @param string $indexName
+     * @param string $locale
+     *
+     * @return string[]
+     */
+    private function getLocalizedIndexNamesFor($indexName, $locale = null)
+    {
+        $adapterIndexNames = $this->adapter->listIndexes();
```

Comment:
if I remove an index from metadata but the data still exist with Zend Lucene listIndexes return the deleted index !!